components/PriceFeedTable.tsx

```diff
@@ -0,0 +1,119 @@
+import { useState, useCallback } from "react";
+import { StyledTd } from "./Table";
+
+interface PriceFeed {
+  ids: string;
+  assetType: string;
+  name: string;
+}
+
+interface TableColumnWidths {
+  assetType: string;
+  name: string;
+  ids: any;
+}
+
+const columnWidths: TableColumnWidths = {
+  assetType: "w-3/10",
+  name: "w-1/5",
+  ids: "w-1/2",
+};
+
+const PriceFeedTable = ({ priceFeeds }: { priceFeeds: PriceFeed[] }) => {
+  const [selectedAssetType, setSelectedAssetType] = useState<string>("All");
+  const [copiedId, setCopiedId] = useState<string | null>(null);
+
+  const assetTypes = [
+    "All",
+    ...Array.from(new Set(priceFeeds.map((feed) => feed.assetType))),
+  ];
+
+  const filteredFeeds =
+    selectedAssetType === "All"
+      ? priceFeeds
+      : priceFeeds.filter((feed) => feed.assetType === selectedAssetType);
+
+  const copyToClipboard = useCallback((text: string) => {
+    navigator.clipboard
+      .writeText(text)
+      .then(() => {
+        setCopiedId(text);
+        setTimeout(() => setCopiedId(null), 2000); // Hide the popup after 2 seconds
+      })
+      .catch((err) => {
+        console.error("Failed to copy: ", err);
+      });
+  }, []);
+
+  return (
+    <div>
+      <div className="mb-4">
+        <label htmlFor="assetTypeFilter" className="mr-2">
+          Filter by Asset Type:
+        </label>
+        <select
+          id="assetTypeFilter"
+          value={selectedAssetType}
+          onChange={(e) => setSelectedAssetType(e.target.value)}
+          className="p-2 border rounded"
+        >
+          {assetTypes.map((type) => (
+            <option key={type} value={type}>
+              {type}
+            </option>
+          ))}
+        </select>
+      </div>
+      <table>
+        <thead>
+          <tr>
+            <th className={columnWidths.assetType}>Asset Type</th>
+            <th className={columnWidths.name}>Name</th>
+            <th className={columnWidths.ids}>Feed ID</th>
+          </tr>
+        </thead>
+        <tbody>
+          {filteredFeeds.map((feed, index) => (
+            <tr key={index}>
```

Comment:
You should use something that's unique and not index based here, e.g. `feed.name`, as the key.  Using the `index` will cause issues when you change the filter because React will try to recycle HTML elements that were used for other feeds when indices change and you'll get hard to debug issues where you'll see stale data in the UI.